### DISCUSSION TITLE
Another php8.2 problem

### DISCUSSION THREAD

**Discussion by wrangster** [geodesicsolutions-community/geocore-community#211] [Q&A] [answered]
Hello again!
I didn't come very far. :D I have a feeling that there will be a few of these php8.2 problems, so bear with me. :) Upgrading from 16.07 went fine, logging in to my old frontuser account went fine, placing listing went fine, but when I try to look at the listing, I get this error:

Template Error: Template file not found!

The template file system/listing_details/mapping_directions_link_google.tpl could not be found in any of the template sets currently loaded.

I could not fint that file anywhere, so I found a file called mapping_link.tpl, copyed it and called it mapping_directions_link_google.tpl

Then I get this error:

TypeError: array_merge(): Argument #1 must be of type array, ADOFetchObj given in /home/2/s/*********/www/*****/classes/browse_display_ad.php:213
Stack trace:
#0 /home/2/s/*********/www/*****/classes/browse_display_ad.php(213): array_merge(Object(ADOFetchObj), Array)
#1 /home/2/s/*********/www/*****/index.php(165): Display_ad->display_classified(2602)
#2 {main}

I don't know if this last error is a result of my template copying, or something I would have met even if that template was the right one. Anyone have a clue? :)

**Comment by wrangster** [comment, marked as answer]
Oh yeah, google (and Stack Overflow) is your friend....! :) Line 210, change this:

$this->browsing_configuration = $this->ad_configuration_data;

to this:

$this->browsing_configuration = json_decode(json_encode($this->ad_configuration_data), true);

This is probably not the best solution, but no I can at least se my listings. :)

**Comment by vicos59** [comment]
Hi @wrangster,

Just as an fyi, we only tested through PHP 8.1.

There are no active developers on the project and PHP 8.2 compatibility is not even on the table at this time. But, feel free to continue posting issues and solutions so they can be incorporated at a future date, as well as for the benefit of others who need to patch their versions.

**Comment by ncrs-admin** [comment]
In Classifieds running PHP 8.0 when I open an ad and click on Contact Seller I get this error instead of the contact form. Any idea how to fix this issue?

TypeError: count(): Argument #1 ($value) must be of type Countable|array, null given in /home/ncrs/public_html/classifiedtest/templates_c/809f9ea78be1fdf28c045ac2a8d418d405503345_0.geo_tset.system_browsing_contact_forms_common_form_top_tpl.php:32
Stack trace:
#0 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_template_resource_base.php(123): content_67a23a591f0657_21680998(Object(Smarty_Internal_Template))
#1 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_template_compiled.php(114): Smarty_Template_Resource_Base->getRenderedTemplateCode(Object(Smarty_Internal_Template))
#2 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(216): Smarty_Template_Compiled->render(Object(Smarty_Internal_Template))
#3 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(385): Smarty_Internal_Template->render()
#4 /home/ncrs/public_html/classifiedtest/templates_c/8e23d27ed778fc5f7db0b91fddf9e2b3adc42017_0.geo_tset.system_browsing_contact_forms_seller_form_tpl.php(31): Smarty_Internal_Template->_subTemplateRender('geo_tset:system...', NULL, NULL, 0, 3600, Array, 0, false)
#5 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_template_resource_base.php(123): content_67a23a591e7e37_36655534(Object(Smarty_Internal_Template))
#6 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_template_compiled.php(114): Smarty_Template_Resource_Base->getRenderedTemplateCode(Object(Smarty_Internal_Template))
#7 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(216): Smarty_Template_Compiled->render(Object(Smarty_Internal_Template))
#8 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatebase.php(238): Smarty_Internal_Template->render(false, 0)
#9 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatebase.php(116): Smarty_Internal_TemplateBase->_execute(Object(Smarty_Internal_Template), NULL, NULL, NULL, 0)
#10 /home/ncrs/public_html/classifiedtest/classes/php5_classes/Template.class.php(485): Smarty_Internal_TemplateBase->fetch()
#11 /home/ncrs/public_html/classifiedtest/classes/geo_smarty_plugins/function.body_html.php(50): geoTemplate::loadInternalTemplate(Array, Object(Smarty_Internal_Template), 'contact_forms/s...', 'system', 'browsing', Array, '', '')
#12 /home/ncrs/public_html/classifiedtest/templates_c/5fd3e62df7228621522118ed8998289acebae845_0.geo_tset.basic_page_tpl.php(36): smarty_function_body_html(Array, Object(Smarty_Internal_Template))
#13 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_template_resource_base.php(123): content_677ff596123aa4_40989960(Object(Smarty_Internal_Template))
#14 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_template_compiled.php(114): Smarty_Template_Resource_Base->getRenderedTemplateCode(Object(Smarty_Internal_Template))
#15 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_internal_template.php(216): Smarty_Template_Compiled->render(Object(Smarty_Internal_Template))
#16 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatebase.php(238): Smarty_Internal_Template->render(false, 1)
#17 /home/ncrs/public_html/classifiedtest/vendor/smarty/smarty/libs/sysplugins/smarty_internal_templatebase.php(134): Smarty_Internal_TemplateBase->_execute(Object(Smarty_Internal_Template), NULL, NULL, NULL, 1)
#18 /home/ncrs/public_html/classifiedtest/classes/php5_classes/View.class.php(436): Smarty_Internal_TemplateBase->display('basic_page.tpl')
#19 /home/ncrs/public_html/classifiedtest/classes/site_class.php(426): geoView->render(6)
#20 /home/ncrs/public_html/classifiedtest/classes/browse_notify_seller.php(192): geoSite->display_page()
#21 /home/ncrs/public_html/classifiedtest/index.php(941): Notify_seller->send_a_message_to_seller_form('3364')
#22 {main}

**Comment by wrangster** [comment]
Hello! Check out these 2 issues about similar errors:
https://github.com/geodesicsolutions-community/geocore-community/issues/227
https://github.com/geodesicsolutions-community/geocore-community/issues/231

**Reply by ncrs-admin** [reply]
Thanks for the suggestions.  I implemented all of them and I still get the error when I try the Contact Seller link.

**Reply by wrangster** [reply]
I did not meen that implementing them would fix your problems, only that you can see how to fix problems related to count. I guess it schould be done in one (or all) of the files mentioned i the bottom of your error message.

**Reply by ncrs-admin** [reply]
Thanks for the feedback.  Never dawned on me.  That is a lot of files, but I will see what I can do.

**Reply by wrangster** [reply]
It's only the 4 files on the bottom :) Search for count, and see if you find something that can be fixed the same way. Good luck! :)
